assets/js/pages/Users/Users.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { noop, find } from 'lodash';
+import { format, parseISO } from 'date-fns';
+
+import Button from '@common/Button';
+import Table from '@common/Table';
+import PageHeader from '@common/PageHeader';
+import Modal from '@common/Modal';
+import Tooltip from '@common/Tooltip';
+import Banner from '@common/Banners/Banner';
+
+const defaultUsers = [];
+
+function getUserByID(users, userID) {
+  return find(users, { id: userID }) || null;
+}
+
+function Users({
+  onDeleteUser = noop,
+  navigate = noop,
+  users = defaultUsers,
+  loading = false,
+}) {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [deleteUserID, setDeleteUserID] = useState(null);
+
+  const usersTableConfig = {
+    pagination: true,
+    usePadding: false,
+    columns: [
+      {
+        title: 'Username',
+        key: 'username',
+        render: (content, item) => (
+          <Link
+            className="text-jungle-green-500 hover:opacity-75"
+            to={`/users/${item.id}/edit`}
+          >
+            {content}
+          </Link>
+        ),
+      },
+      {
+        title: 'Full Name',
+        key: 'fullname',
+      },
+      {
+        title: 'Email',
+        key: 'email',
+      },
+      {
+        title: 'Status',
+        key: 'enabled',
+        render: (content, item) => (
+          <span>{item.enabled ? 'Enabled' : 'Disabled'}</span>
+        ),
+      },
+      {
+        title: 'Created',
+        key: 'created_at',
+        render: (content, item) => (
+          <span> {format(parseISO(item.created_at), 'MMMM dd, yyyy')} </span>
+        ),
+      },
+      {
+        title: 'Actions',
+        key: 'actions',
+        render: (content, item) => (
+          <div>
+            <Tooltip
+              content="Admin user can not be deleted"
+              isEnabled={item.id === 1}
+            >
+              <Button
+                className="text-red-500"
+                size="small"
+                type="transparent"
+                disabled={item.id === 1}
+                onClick={() => {
+                  setModalOpen(true);
+                  setDeleteUserID(item.id);
+                }}
+              >
+                Delete
+              </Button>
+            </Tooltip>
+          </div>
+        ),
+      },
+    ],
+  };
+
+  const user = getUserByID(users, deleteUserID);
```

Comment:
Please use a proper state, this is not a good thing for the component lifecycle

Reply:
Removing this helper, added selectedUser state, with this the whole component can work with user data  and this becomes obsolete :)